x-pack/plugins/maps/public/connected_components/timeslider/timeslider_popover.tsx

```diff
@@ -0,0 +1,196 @@
+          setLabel(i18n.translate('xpack.maps.timeslider.autoPeriod', { defaultMessage: 'Auto' }));
+        }}
+      >
+        {i18n.translate('xpack.maps.timeslider.autoPeriod', {
+          defaultMessage: 'Auto',
+        })}
+      </EuiContextMenuItem>
+    );
+
+    return filteredPeriods;
+  };
+
+  const onFormSubmit = (e: React.SyntheticEvent) => {
+    e.preventDefault();
+    const target = e.target as typeof e.target & {
+      time: { value: string };
+      type: { value: string };
+    };
+    setShowErrors(false);
+    if (target.time.value && getCustomInterval(target) < max - min) {
+      props.customIntervalHandler(e);
+      setLabel(getCustomLabel(target));
+      setIsOpen(false);
+    } else {
+      setShowErrors(true);
+    }
+  };
+  const error = i18n.translate('xpack.maps.timeslider.invalidSelectedInterval', {
+    defaultMessage:
+      'The selected interval cannot be empty or exceed the global time window. Please reduce the interval or change the global time window.',
+  });
+
+  const intervalForm = (
+    <>
+      <EuiPopoverTitle>
+        {i18n.translate('xpack.maps.timeslider.intervalsTitle', {
+          defaultMessage: 'Intervals',
+        })}
+      </EuiPopoverTitle>
+      <EuiForm
+        className="mapTimeslider_form"
+        component="form"
+        onSubmit={onFormSubmit}
+        isInvalid={showErrors}
+        error={error}
+      >
+        <EuiFlexGroup gutterSize="s" justifyContent="center">
+          <EuiFlexItem grow={false}>
+            <EuiFormRow>
+              <EuiFieldNumber name="time" compressed min={1} max={999} />
+            </EuiFormRow>
+          </EuiFlexItem>
+          <EuiFlexItem grow={false}>
+            <EuiFormRow>
+              <EuiSelect name="type" compressed options={filteredOptions} />
+            </EuiFormRow>
+          </EuiFlexItem>
+          <EuiFlexItem grow={false}>
+            <EuiFormRow>
+              <EuiButton type="submit" size="s">
+                {i18n.translate('xpack.maps.timeslider.applyButton', {
+                  defaultMessage: 'Apply',
+                })}
+              </EuiButton>
+            </EuiFormRow>
+          </EuiFlexItem>
+        </EuiFlexGroup>
+      </EuiForm>
+      <EuiHorizontalRule margin="xs" />
+    </>
+  );
+
+  return (
+    <div className="mapTimeslider__timeRange">
+      {renderFilteredPeriods()?.length > 1 && (
+        <EuiPopover
+          id="metricsPopover"
+          isOpen={isOpen}
+          closePopover={() => setIsOpen(false)}
+          panelPaddingSize="s"
+          ownFocus
+          anchorPosition="upCenter"
+          button={
+            <EuiButton
+              size="s"
+              className="mapTimeslider__euiButton-controls"
+              onClick={() => {
+                setIsOpen((x) => !x);
+                setShowErrors(false);
+              }}
+            >
+              {label}
+            </EuiButton>
+          }
+        >
+          {intervalForm}
```

Comment:
Is the intervalForm needed in this PR? I thought we had decided to scope the first iteration to just selecting some auto generated ranges. I think we need more design work on this form and the first step of just showing a few ranges is a good place to start untill we get better input from design

Reply:
+1 

I'd favor keeping this first PR small, and just introduce the "Commonly used" section.

We can introduce the "Intervals" in a separate form.